paasta_tools/cli/cmds/status.py

```diff
@@ -2407,9 +2407,14 @@ def paasta_status(args) -> int:
 
     with concurrent.futures.ThreadPoolExecutor(max_workers=20) as executor:
         tasks = [executor.submit(t[0], **t[1]) for t in tasks]  # type: ignore
-        for future in concurrent.futures.as_completed(tasks):  # type: ignore
-            return_code, output = future.result()
-            return_codes.append(return_code)
+        try:
+            for future in concurrent.futures.as_completed(tasks):  # type: ignore
+                return_code, output = future.result()
+                return_codes.append(return_code)
+        except KeyboardInterrupt:
+            executor._threads.clear()  # type: ignore
```

Comment:
nit: giving an explanation for the `type: ignore`/private member access would probably be a good idea
```suggestion
            # ideally we wouldn't need to reach into `ThreadPoolExecutor`
            # internals, but so far this is the best way to stop all these
            # threads until a public interface is added
            executor._threads.clear()  # type: ignore
```